packages/react-dom/src/server/ReactPartialRendererHooks.js

```diff
@@ -64,6 +70,20 @@ let isInHookUserCodeInDev = false;
 // In DEV, this is the name of the currently executing primitive hook
 let currentHookNameInDev: ?string;
 
+export function getHookInternals() {
+  return {
+    currentlyRenderingComponent,
+    firstWorkInProgressHook,
+    workInProgressHook,
+  };
```

Comment:
What about `renderPhaseUpdates` and others? I think those would break too.

Reply:
I added only variables which were different before modification in finishHooks. Should I add others?

Reply:
`renderPhaseUpdates` gets set to `null` there. So I think it might break if you don't.

One thing you can try is writing a test that triggers writes and reads to each of them. (E.g. by breaking them and finding existing tests that cover this code.) Then write a test that the same pattern works with nested render calls.

Reply:
Seems like this pattern breaks when adding second response.

Reply:
Have you figured out why?

Reply:
Test passes, but there is no isolation of variables. Because renderers are asynchronous, one of the renderers rewrites `workInProgressHook` to undefined in one of the `createWorkInProgressHook` calls. Perhaps its because during execution the wrong `workInProgressHook` used. When `hookInternals` are initialized for the second time, they are empty objects, despite comment in `prepareToUseHooks` which says they should be the `null` values (and `workInProgressHook` is of the Hook type in the second `prepareToUseHooks` call).